### PR TITLE
Branch event sourcing

### LINKED ISSUE #145: Convert branch persistence to event sourcing
Currently some of our data is stored using events, and some of it is in a relational database. This task is to store branches using events.

### PR DESCRIPTION
**Pull request by camjackson**
This resolves #145.

- This should be merged *after* #137 and #144
- Trying very hard to work in much small commits:
    - All the tests should be passing at each step
    - But some of the commits might leave the app half-working. E.g. A single commit might replace database insertions with event publishing, but not actually handle those events yet.
    - Therefore the commits for this PR should probably be squashed when the PR is merged, otherwise there will be points in the tree where things don't work.

### CODE REVIEW

**Comment by camjackson** [comment]
This PR is ready for review now! It's already deployed to our staging environment, so for anyone who wants to manually test it, if you log into the admin app, you should be able to create/edit/delete branches. One functional change here is that the API no longer allows you to delete a branch that has members in it, but this is not handled gracefully by the frontend. That's something we should fix, but for now you'll just get a generic error if you try to delete a branch with members.

File: backend/spec/integration/membersIntegrationSpec.js [hunk 1]
```diff
@@ -131,19 +132,24 @@ describe('MemberIntegrationTests', () => {
...
-            .then(integrationTestHelpers.authenticateBranchAdmin(agent))
-            .then(getMembers(agent, browserState))
-            .then(setState(browserState, 'members')));
+    beforeEach(() => (
+      integrationTestHelpers.resetDatabase()
+        .then(integrationTestHelpers.createBranch)
+        .then(setState(browserState, 'branch'))
+        .then(branch => (
+          integrationTestHelpers.createBranchAdmin(branch)
+            .then(() => branch)
```
Comment (camjackson) [inline backend/spec/integration/membersIntegrationSpec.js]: This extra nested `then` is where the actual fix is. Previously the promise chain would sometimes move on before the branch admin had been created.

**Comment by camjackson** [comment]
I've just seen how huge the all-in-one diff is for this PR. Probably a good idea to review each commit one by one, they're a bit more digestible that way.

File: backend/package.json [hunk 2]
```diff
@@ -13,7 +13,7 @@
     "migrate": "sequelize db:migrate",
     "e2e-seed": "NODE_ENV=test node bin/e2e-seed.js",
     "pretest": "NODE_ENV=test npm run migrate && node bin/test-seed.js",
-    "test": "NODE_ENV=test mocha --reporter dot --check-leaks --timeout 10000 -r spec/specGlobals --recursive spec",
+    "test": "NODE_ENV=test mocha --reporter dot --check-leaks --timeout 10000 -r spec/specGlobals --recursive spec/unit",
```
Comment (camjackson) [inline backend/package.json]: Integration tests were too hard to fix at this point. They come back later.

**Comment by pameck** [comment]
Review notes:

- Sequelize is coming to an end!! Yessss!
- Ran tests, all good.
- Ran the app, it saves branches correctly.
- I'm updating the `README` with the manual seeding step.
- There is a racing condition: When the UI creates the branch, it tries to fetch its details right away, but the branch might not have been store at that point, because it is going from `kinesis` to the `forwarder-lambda` and back to the core before it is saved to the redux store. We will address that in a separate issue.

**Pull request merged**